### ISSUE TITLE
Initial requirements

### ISSUE THREAD

**Issue by aulneau** [fungible-systems/stacks-apps-registry#1]
I've been thinking about how we as a community can work together to create a more rich experience when looking at activity on the network. When you look at the Hiro explorer, you see a bunch of transactions, blocks, and addresses. It's really hard for folks to tie together all these contracts and associate them with a given app (gamma, alex, etc).

I'm proposing that we build out an app registry that allows the community to connect all these contracts together, provide metadata, and make it more human readable.

With that being said, this is the initial overview I've thought of around features:

- concept of "an app"
- tie an app to a BNS name
- connect many contracts together under the app
- provide metadata for app (logo, description, links etc)
- perhaps some signed attestation of ownership (verified apps, etc)
- perhaps metadata for contracts, their functions, and descriptions etc
- this data can be indexed and exposed as an API anyone can use freely
- anything else?

**Comment by LNow** [comment]
Probably some data should be stored on-chain, and some off-chain - the same way NFT's are handled.

One of the main question I have is who should be responsible for creating new "app", as well as who should have the possibility to change app settings?

**Comment by aulneau** [comment]
> Probably some data should be stored on-chain, and some off-chain - the same way NFT's are handled.

Agree, I'm thinking of building out a small intake app that would force a given standard for the metadata, and auto upload it to ipfs or something like that.

> One of the main question I have is who should be responsible for creating new "app", as well as who should have the possibility to change app settings?

Yeah, good question. We can used sign messages to accomplish this maybe? or have some validation mechanism in place, like generate a random string, have them put it as a header on the domain of the app or something like that. What do you think?

**Comment by LNow** [comment]
If you want registry to be trusted then IMHO someone needs to control what it stores, therefore I would start with fully centralized approach, where only registry contract deployer is able to create and update app settings.

**Comment by aulneau** [comment]
I agree, might be good to have the option of adding alternative addresses with admin rights too, or rather "moderation" powers

**Comment by spliffbtc** [comment]
I like this idea. Let me know if/how I can help.
